activerecord/test/cases/associations/has_many_through_associations_test.rb

```diff
@@ -1252,4 +1255,56 @@ def test_incorrectly_ordered_through_associations
       )
     end
   end
+
+  def test_belongs_to_autosave_true_works_for_one_association
+    cs180 = Seminar.new(name: "CS180")
+    fall = Session.new(name: "Fall",
+                        start_date: "2017-09-01",
+                        end_date: "2017-12-31")
+    fall.sections << (cs180a = cs180.sections.build(short_name: "A"))
+
+    assert cs180.save
+
+    cs180.reload
+    fall.reload
+    assert_equal [cs180a].sort_by(&:id), cs180.sections.sort_by(&:id)
+    assert_equal [cs180a].sort_by(&:id), fall.sections.sort_by(&:id)
+  end
+
+  def test_belongs_to_autosave_true_fails_for_two_associations
```

Comment:
When I first observed the issue, I noticed that it didn't seem to happen if there was only one element in the association. As I recall, I decided to write a test that would confirm that was the case. So, short answer, `test_belongs_to_autosave_true_works_for_one_association` should pass without the fix included in this PR. `test_belongs_to_autosave_true_fails_for_two_associations` should fail without the fix in this PR.